textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/fuzzytokenmatcher/FuzzyTokenMatcher.java

```diff
@@ -28,16 +36,61 @@ public FuzzyTokenMatcher(IPredicate predicate) {
     
     @Override
     public void open() throws DataFlowException {
-    	return;
+    	try {
+            sourceOperator.open();
+            attributeList = predicate.getAttributeList();
+            threshold = predicate.getThreshold();
+            queryTokens = predicate.getQueryTokens();
+    	} catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+    	}
     }
 
     @Override
     public ITuple getNextTuple() throws DataFlowException {
-	return null;
+		try {
+		    ITuple sourceTuple = sourceOperator.getNextTuple();
+		    if (sourceTuple == null || !this.predicate.getisSpanInformationAdded())
+		        return sourceTuple;
+            
+		    int schemaIndex = sourceTuple.getSchema().getIndex(SchemaConstants.SPAN_LIST_ATTRIBUTE.getFieldName());
+		    List<Span> resultSpans =
+                    (List<Span>)sourceTuple.getField(schemaIndex).getValue();
+            
+		    for(int attributeIndex = 0; attributeIndex < attributeList.size(); attributeIndex++) {
+		        String fieldName = attributeList.get(attributeIndex).getFieldName();
+		        IField field = sourceTuple.getField(fieldName);
+                
+		        if (field instanceof TextField) {
+		            int tokensMatched = 0;
+		            List<Span> attributeSpans = new ArrayList<>();
+		            for (Span span : resultSpans) {
+		                if (span.getFieldName().equals(fieldName)) {
+		                    attributeSpans.add(span);
+		                    if (queryTokens.contains(span.getKey()))
+		                        tokensMatched++;
+		                }
+		            }
+		            if (tokensMatched < threshold) {
```

Comment:
Let's say resultSpans currently has 15 spans (10 from A & 5 from B). It is an IField of sourcetuple. So instead of removing B's spans, if we wish to to construct a temporary variable and add A's spans to it (which is of the same order as above), then we also need to change the sourcetuple to accommodate this new variable and eliminate the older resultSpans. I feel either way is good.